internal/handler/handler.go

```diff
@@ -0,0 +1,99 @@
+package handler
+
+import (
+	"fmt"
+	"github.com/Kuart/metric-collector/internal/metric"
+	"github.com/Kuart/metric-collector/internal/storage"
+	"github.com/Kuart/metric-collector/internal/template"
+	"github.com/go-chi/chi/v5"
+	"net/http"
+	"strconv"
+)
+
+const (
+	notIntError     = "value is not int type"
+	notFloatError   = "value is not float type"
+	metricTypeError = "metric type not implemented"
+	metricNotFound  = "metric \"%s\" is not found"
+)
+
+func SetRoutes(r *chi.Mux) {
+	r.Get("/value/{type}/{name}", MetricValueHandler)
+	r.Route("/update", func(r chi.Router) {
+		r.Post("/{type}/{name}/{value}", UpdateHandler)
+	})
+	r.Get("/", MetricsPageHandler)
+}
+
+func UpdateHandler(w http.ResponseWriter, r *http.Request) {
+	metricType := chi.URLParam(r, "type")
+
+	switch metricType {
+	case metric.GaugeTypeName:
+		GaugeHandler(w, r)
+	case metric.CounterTypeName:
+		CounterHandler(w, r)
+	default:
+		http.Error(w, metricTypeError, http.StatusNotImplemented)
+	}
+}
+
+func CounterHandler(w http.ResponseWriter, r *http.Request) {
+	name, valueString := chi.URLParam(r, "name"), chi.URLParam(r, "value")
+
+	value, err := strconv.ParseInt(valueString, 10, 64)
+
+	if err != nil {
+		http.Error(w, notIntError, http.StatusBadRequest)
+	} else {
+		storage.CounterUpdate(name, value)
+		w.WriteHeader(http.StatusOK)
+	}
```

Comment:
можно избавляться  от лишних ветвлений, тогда код чуточку удобнее читать
```suggestion
                   return /// если ошибку возвращаем, то смысла нам дальше идти по коду не надо
	}
	storage.CounterUpdate(name, value)
	w.WriteHeader(http.StatusOK)
```

Reply:
в принципе по коду дальше можно тоже применять, не буду акцентировать на этом внимание в этом ревью